### DISCUSSION TITLE
Issues with zeus login/preview

### DISCUSSION THREAD

**Discussion by SwitchAxe** [zepp-health/zeppos-forum#389] [Q&A]
Hi, on my Linux (openSUSE Tumbleweed, KDE Plasma on Wayland) OS, `zeus login` and `zeus preview` don't work. trying to execute them tells me that a window has been opened in my default browser (which would be firefox) but it's not the case. How can i get around this? is this a bug or am i doing something wrong?  
pic for details:
![immagine](https://github.com/user-attachments/assets/e0e42919-64da-4aea-976c-6e50bdb44d30)

**Comment by silver-zepp** [comment]
Hey @SwitchAxe, thanks for the details. We'll look into this issue and let you know as soon as we have a solution.

**Reply by SwitchAxe** [reply]
let me know if you need any more details/logs that could help in fixing this!

**Reply by zeppos-hx** [reply]
I found that this is a bug in the built-in xdg-open of a zeus dependency package, will try to fix it soon, thank you so much

**Comment by ashishbhate** [comment]
I've run into this issue as well. I was able to workaround the issue by printing the URL that the code tries to open, and then opening that in my browser manually.

1. Locate where the `zeus-cli` is installed. 
In my case it's `~/.nvm/versions/node/v20.17.0/lib/node_modules/@zeppos/zeus-cli/`

2. Locate the file that handles login. 
In my case it's at `~/.nvm/versions/node/v20.17.0/lib/node_modules/@zeppos/zeus-cli/private-modules/zeppos-app-utils/dist/modules/login.js`. 

3. Print the URL. The code is obfuscated but easy to understand. 
It's in `case 2` (line 119) of the `login` function (line 86). Its possible the line numbers will be different for you.
I added the following before the return statement for the case. Again, this might be different for you because of code obfuscation 
```JS
                index_1.logger.info("".concat(loginConfig.LOGIN_URL, "?").concat(qs_1["default"].stringify({
                        project_name: loginConfig.PROJECT_NAME,
                        platform_app: loginConfig.PLATFORM_APP,
                        project_redirect_uri: loginCallbackUrl
                    })));
```

**Reply by qroft** [reply]
I followed all of your steps, but it is still not working.
All this comes inside "case 2" right? 
Where do you see the obfuscated URL?
Thanks in advance.

**Comment by zeppos-hx** [comment]
The latest version [v1.6.1](https://www.npmjs.com/package/@zeppos/zeus-cli/v/1.6.1) was released, and fixed this issue. Thank you for the details @SwitchAxe and @ashishbhate

**Reply by qroft** [reply]
Unfortunately i am having the same problem on my Linux Mint.
The thing is, that on the first run, it worked like a charm.
I thought that it might be because of the "saved password" that i allowed Firefox to store.
But even after removing the password from the saved passwords, it still does not work.

Also i have always to start both options as sudo

**Reply by qroft** [reply]
I am having a strange issue now as there seems to be a difference wether i have installed it as sudo or not.

If i in the terminal run zeus preview, then instantly can i select the device.
After selecting the device, i get a permission error that i am not allowed to execute the code.

So i ran sudo zeus preview and now, without asking for the target device, i get "The login page has been opened...."
But nothing happens.
